Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/UserController.cs

```diff
@@ -204,192 +238,265 @@ public IActionResult GetRole()
         * @apiParamExample {json} Request-Example:
-        *      
-        *        {
-        *             "OldPassword":"OldPassword",
-        *             "NewPassword":"NewPassword",
-        *             "ConfirmPassword":"ConfirmPassword"
+        * {
+        *     "OldPassword":"OldPassword",
+        *     "NewPassword":"NewPassword",
+        *     "ConfirmPassword":"ConfirmPassword"
         *            
-        *        }      
+        * }      
         * @apiSuccessExample {json} Success-Response:
         * HTTP/1.1 200 OK 
         * {
-        *     "description":"Change Password"
+        *     "newPassword":"newPassword"
+        * }
+        * @apiError UserNotFound the slack user was not found.
+        * @apiErrorExample {json} Error-Response:
+        * HTTP/1.1 404 Not Found
+        * {
+        *   "error": "UserNotFound"
+        * }
+        * @apiError UserNotFound the slack user was not found.
+        * @apiErrorExample {json} Error-Response:
+        * HTTP/1.1 404 Not Found
+        * {
+        *   "error": "UserNotFound"
         * }
         */
         [HttpPost]
-        [Route("changepassword")]
-        [AllowAnonymous]
-        [Authorize(Roles = "Admin,Employee")]
-        public async Task<IActionResult> ChangePassword([FromBody] ChangePasswordViewModel passwordModel)
+        [Route("password")]
+        [Authorize]
+        public async Task<IActionResult> ChangePasswordAsync([FromBody] ChangePasswordViewModel passwordModel)
         {
             try
             {
                 var user = await _userManager.GetUserAsync(User);
                 passwordModel.Email = user.Email;
-                string response = await _userRepository.ChangePassword(passwordModel);
+                string response = await _userRepository.ChangePasswordAsync(passwordModel);
                 return Ok(new { response });
             }
-            catch (UserNotFound ex)
+            catch (UserNotFound)
             {
                 return NotFound();
             }
-            catch (Exception ex)
-            {
-                return BadRequest();
-            }
-        }
 
+        }
 
+        /**
+       * @api {post} api/users/:password/available 
+       * @apiVersion 1.0.0
+       * @apiName CheckPasswordAsync
+       * @apiGroup User
+       * @apiParam {string} password  User OldPassword
+       * @apiParamExample {json} Request-Example:
+       * {
+       *     "password":"OldPassword123"
+       * }      
+       * @apiSuccessExample {json} Success-Response:
+       * HTTP/1.1 200 OK 
+       * {
+       *     true
+       * }
+       */
         [HttpGet]
-        [Route("checkOldPasswordIsValid/{oldPassword}")]
-        [Authorize(Roles = "Admin,Employee")]
-        public async Task<ActionResult> CheckOldPasswordIsValid(string oldPassword)
+        [Route("{password}/available")]
+        [Authorize]
+        public async Task<ActionResult> CheckPasswordAsync(string password)
```

Comment:
This method used to check old password is exists or not, when user enters old password at change password functionality.